mne/label.py

```diff
@@ -2020,6 +2036,147 @@ def read_labels_from_annot(subject, parc='aparc', hemi='both',
     return labels
 
 
+def _check_labels_subject(labels, subject, name):
+    _validate_type(labels, (list, tuple), 'labels')
+    for label in labels:
+        _validate_type(label, Label, 'each entry in labels')
+        if subject is None:
+            subject = label.subject
+        if subject is not None:
+            if subject != label.subject:
+                raise ValueError('Got multiple values of %s: %s and %s'
+                                 % (name, subject, label.subject))
+    if subject is None:
+        raise ValueError('if label.subject is None for all labels, '
+                         '%s must be provided' % name)
+    return subject
+
+
+@verbose
+def morph_labels(labels, subject_to, subject_from=None, subjects_dir=None,
+                 surf_name='white', verbose=None):
+    """Morph a set of labels.
+
+    This is useful when morphing a set of non-overlapping labels (such as those
+    obtained with :func:`read_labels_from_annot`) from one subject to
+    another.
+
+    Parameters
+    ----------
+    labels : list
+        The labels to morph.
+    subject_to : str
+        The subject to morph labels to.
+    subject_from : str | None
+        The subject to morph labels from. Can be None if the labels
+        have the ``.subject`` property defined.
+    subjects_dir : string, or None
+        Path to SUBJECTS_DIR if it is not set in the environment.
+    surf_name : str
+        Surface used to obtain vertex locations, e.g., 'white', 'pial'
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Returns
+    -------
+    labels : list
+        The morphed labels.
+
+    See Also
+    --------
+    read_labels_from_annot
+    mne.Label.morph
+
+    Notes
+    -----
+    This does not use the same algorithm as Freesurfer, so the results
+    morphing (e.g., from ``'fsaverage'`` to your subject) might not match
+    what Freesurfer produces during ``recon-all``.
+
+    .. versionadded:: 0.18
+    """
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    subject_from = _check_labels_subject(labels, subject_from, 'subject_from')
+    mmaps = read_morph_map(subject_from, subject_to, subjects_dir)
+    vert_poss = [_load_vert_pos(subject_to, subjects_dir, surf_name, hemi,
+                                mmap.shape[0])
+                 for hemi, mmap in zip(('lh', 'rh'), mmaps)]
+    idxs = [_sparse_argmax(mmap, axis=1) for mmap in mmaps]
+    out_labels = list()
+    values = filename = None
+    for label in labels:
+        li = dict(lh=0, rh=1)[label.hemi]
```

Comment:
isn't `0 if label.hemi == 'lh' else 1` much nicer?

Reply:
(hemi is never neither lh nor rh right?)

Reply:
It's nicer in some ways, yes, but in others this is nicer (e.g., it answers the parenthetical question you just asked)

Reply:
ok ok, I'll stop commenting on your PRs :D

Arguably, if hemi is neither lh nor rh, the dict approach will crash in a *slightly less* confusing manner. But you have to admit you're abusing dict here.

Reply:
> the dict approach will crash in a slightly less confusing manner.

Actually it might not crash at all, and silently do something bad...

But yes I agree I am abusing dict by using such a powerful class to do what is a compressed if/elif/else-crash :) And I'm sure elsewhere we do `0 if hemi == 'lh' else 1` and it's fine by me either way

Reply:
> ok ok, I'll stop commenting on your PRs :D

I don't mind style discussions :)